wolfcrypt/src/port/Renesas/renesas_tsip_util.c

```diff
@@ -143,6 +151,279 @@ static uint32_t GetTsipCipherSuite(
+    byte*   peerkey = NULL;
+    word32  qxLen = sizeof(qx), qyLen = sizeof(qy);
+    TsipUserCtx*   userCtx;
+
+    WOLFSSL_MSG(">> tsip_ServerKeyExVerify");
+
+    /* sanity check */
+    if (ssl == NULL || sig == NULL || ctx == NULL)
+        return ret;
+    
+    userCtx = (TsipUserCtx*)ctx;
+
+    /* export public peer public key */
+    ret = wc_ecc_export_public_raw(ssl->peerEccKey, qx, &qxLen, qy, &qyLen);
+
+    if (ret != 0){
+        WOLFSSL_MSG("failed to export peer ecc key");
+        return ret;
+    }
+    /* make peer ecc key data for SCE */
+    /* 0padding(24bit) || 04(8bit) || Qx(256bit) || Qy(256bit) */
+    peerkey = (byte*)XMALLOC((3 + 1 + qxLen + qyLen), NULL,
+                                                 DYNAMIC_TYPE_TMP_BUFFER);
+    if (peerkey == NULL) {
+        WOLFSSL_MSG("failed to malloc ecc key");
+        return WOLFSSL_FAILURE;
+    }
+    
+    XMEMSET(peerkey, 0, (3 + 1 + qxLen + qyLen));
+    peerkey[3] = ECC_POINT_UNCOMP;
+    XMEMCPY(&peerkey[4], qx, qxLen);
+    XMEMCPY(&peerkey[4+qxLen], qy, qyLen);
+    
+    /* 0 : RSA 2048bit, 1 : Reserved, 2 : ECDSA P-256 */
+    if ((ret = tsip_hw_lock()) == 0) {
+        ret = R_TSIP_TlsServersEphemeralEcdhPublicKeyRetrieves(
+            type,
+            (uint8_t*) ssl->arrays->clientRandom,
+            (uint8_t*) ssl->arrays->serverRandom,
+            (uint8_t*) peerkey,
+            (uint8_t*) sig,
+            (uint32_t*)ssl->peerSceTsipEncRsaKeyIndex,
+            (uint32_t*)userCtx->encrypted_ephemeral_ecdh_public_key);
+    
+        if (ret !=TSIP_SUCCESS) {
+            WOLFSSL_MSG("R_TSIP_TlsServersEphemeralEcdhPublicKeyRetrieves failed");
+        } else {
+            ret = WOLFSSL_SUCCESS;
+        }
+    
+        tsip_hw_unlock();
+    }
+    else {
+        WOLFSSL_MSG("Failed to lock tsip hw");
+    }
+
+    XFREE(peerkey, 0, DYNAMIC_TYPE_TMP_BUFFER);
+
+    WOLFSSL_MSG("<< tsip_ServerKeyExVerify");
+    return ret;
+}
+/*
+ *  return 0 on success
+ */
+int wc_tsip_RsaVerify(
+        WOLFSSL* ssl,
+        byte* sig,      word32 sigSz,
+        byte** out,
+        const byte* key,
+        word32 keySz,
+        void* ctx)
+{
+    int ret;
+    
+    WOLFSSL_MSG(">> tsip_RsaVerify");
+    
+    if (tsip_usable(ssl, 0))
+        ret = tsip_ServerKeyExVerify(0, ssl, sig, sigSz, ctx);
+    else
+        ret = CRYPTOCB_UNAVAILABLE;
+
+    if (ret == WOLFSSL_SUCCESS)
+        ret = 0;
+
+    WOLFSSL_MSG("<< tsip_RsaVerify");
+    return ret;
+}
+/*
+ *  return 0 on success
+ */
+int wc_tsip_EccVerify(
+        WOLFSSL*  ssl, 
+        const byte* sig,    word32  sigSz,
+        const byte* hash,   word32  hashSz,
+        const byte* key,    word32  keySz,
+              int*  result, void*   ctx)
+{
+    int         ret = WOLFSSL_FAILURE;
+    uint8_t*    sigforSCE;
+    uint8_t*    pSig;
+    const byte  rs_size = R_TSIP_ECDSA_DATA_BYTE_SIZE/2;
+    byte        offset = 0x3;
+    
+    WOLFSSL_MSG(">> wc_tsip_EccVerify");
```

Comment:
I notice that you have used `WOLFSL_MSG()` with `>>` for function enter and `<<` for function exit.  Should we just change these to use `WOLFSSL_ENTER` and `WOLFSSL_LEAVE` debug macros instead of using `>>` and `<<`?  This applies throughout the entire TSIP work where these are used.

Reply:
Please address

Reply:
Replaced them with WOLFSSL_ENTER/WOLFSSL_LEAVE.